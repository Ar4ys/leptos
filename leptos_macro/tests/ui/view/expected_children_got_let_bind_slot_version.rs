use leptos::*;

#[slot]
struct Then {
    children: Children,
}

#[component]
fn Component(then: Then) -> impl IntoView {
    _ = then;
}

fn main() {
    view! {
        <Component>
            <Then slot let:a>
                <p />
            </Then>
        </Component>
    };
}
