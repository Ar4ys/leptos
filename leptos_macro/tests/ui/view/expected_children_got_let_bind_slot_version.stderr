error[E0308]: mismatched types
  --> tests/ui/view/expected_children_got_let_bind_slot_version.rs:17:17
   |
17 |                 <p />
   |                 ^^^^^ expected `Box<dyn FnOnce() -> Fragment>`, found closure
   |
   = note: expected struct `Box<(dyn FnOnce() -> Fragment + 'static)>`
             found closure `{closure@$DIR/tests/ui/view/expected_children_got_let_bind_slot_version.rs:17:17: 17:22}`
